src/Microsoft.TemplateEngine.Cli/TableOutput/TemplateGroupDisplay.cs

```diff
@@ -26,13 +28,25 @@ internal static class TemplateGroupDisplay
         /// <param name="language">language from the command input.</param>
         /// <param name="defaultLanguage">default language.</param>
         /// <returns></returns>
-        internal static IReadOnlyList<TemplateGroupTableRow> GetTemplateGroupsForListDisplay(IEnumerable<ITemplateInfo> templateList, string language, string defaultLanguage)
+        internal static IReadOnlyList<TemplateGroupTableRow> GetTemplateGroupsForListDisplay(IEnumerable<ITemplateInfo> templateList, string? language, string? defaultLanguage)
         {
             List<TemplateGroupTableRow> templateGroupsForDisplay = new List<TemplateGroupTableRow>();
-            IEnumerable<IGrouping<string, ITemplateInfo>> groupedTemplateList = templateList.GroupBy(x => x.GroupIdentity, x => !string.IsNullOrEmpty(x.GroupIdentity), StringComparer.OrdinalIgnoreCase);
-            foreach (IGrouping<string, ITemplateInfo> grouping in groupedTemplateList)
+            IEnumerable<IGrouping<string?, ITemplateInfo>> groupedTemplateList = templateList.GroupBy(x => x.GroupIdentity, x => !string.IsNullOrEmpty(x.GroupIdentity), StringComparer.OrdinalIgnoreCase);
+            foreach (IGrouping<string?, ITemplateInfo> templateGroup in groupedTemplateList)
```

Comment:
`string` in the `IGrouping` couldn't be `null` because we have a `!string.IsNullOrEmpty` above, right? If this is just to make the compiler happy, maybe we can use `!` somewhere?

Reply:
no, string can be null here when the template doesn't have group identity defined.  `GroupBy` is a custom implementation here: https://github.com/dotnet/templating/blob/cffb87d379e0c127a920a4be96177f0a41702fe8/src/Microsoft.TemplateEngine.Utils/ListExtensions.cs#L13